cmd/sync/sync.go

```diff
@@ -42,36 +47,80 @@ func NewSyncApp() *SyncApp {
 func (app *SyncApp) Cleanup() {
 
 }
+
 func (app *SyncApp) Start(cmd *cobra.Command, args []string) {
 	// start p2p services
 	lg := log.New("sync_test", "", "")
-	lg.Info("Initializing P2P services")
+	lg.Info("Start!!!!!!!!!!!!!!!!!!!!!!!!!!!")
 	swarm, err := p2p.New(cmdp.Ctx, app.Config.P2P)
 
 	if err != nil {
 		panic("something got fudged while creating p2p service ")
 	}
 
-	conf := sync.Configuration{SyncInterval: 1 * time.Second, Concurrency: 4, LayerSize: int(5), RequestTimeout: 100 * time.Millisecond}
-	gTime, err := time.Parse(time.RFC3339, app.Config.GenesisTime)
-	ld := time.Duration(app.Config.LayerDurationSec) * time.Second
-	clock := timesync.NewTicker(timesync.RealClock{}, ld, gTime)
-	msh := mesh.NewPersistentMesh(app.Config.DataDir, sync.ConfigTst(), &sync.MeshValidatorMock{}, sync.MockState{}, sync.AtxDbMock{}, lg)
-	defer msh.Close()
-	if lyr, err := msh.GetLayer(100); err != nil || lyr == nil {
-		lg.Error("could not load layers from disk ...   shutdown", err)
+	iddbstore, err := database.NewLDBDatabase(app.Config.DataDir+"ids", 0, 0)
+	if err != nil {
+		lg.Error("error: ", err)
 		return
 	}
-	lg.Info("woke up with %v layers in meshDB ", 100)
-	app.sync = sync.NewSync(swarm, msh, sync.BlockValidatorMock{}, conf, clock.Subscribe(), lg)
-	if err != nil {
-		log.Error("Error starting p2p services, err: %v", err)
-		panic("Error starting p2p services")
+	idStore := activation.NewIdentityStore(iddbstore)
+	npstCfg := nipst.PostParams{
+		Difficulty:           5,
+		NumberOfProvenLabels: 10,
+		SpaceUnit:            1024,
+	}
+	validator := nipst.NewValidator(npstCfg)
+	mshDb := mesh.NewPersistentMeshDB(app.Config.DataDir, lg.WithOptions(log.Nop))
+	atxdb := activation.NewActivationDb(database.NewMemDatabase(), idStore, mshDb, uint64(app.Config.CONSENSUS.LayersPerEpoch), validator, lg.WithName("atxDb").WithOptions(log.Nop))
+	msh := mesh.NewMesh(mshDb, atxdb, sync.ConfigTst(), &sync.MeshValidatorMock{}, &sync.MockState{}, lg.WithOptions(log.Nop))
+	defer msh.Close()
+
+	conf := sync.Configuration{
+		SyncInterval:   1 * time.Second,
+		Concurrency:    4,
+		LayerSize:      int(100),
+		RequestTimeout: 500 * time.Millisecond,
+	}
+
+	ch := make(chan types.LayerID, 1)
+	app.sync = sync.NewSync(swarm, msh, sync.BlockValidatorMock{}, conf, ch, lg.WithName("syncer"))
+	ch <- 101
```

Comment:
this method has a lot of hard-coded values. it will be easier to understand if they'll be assigned to variables with meaningful names before using them, it will also be easier to change them in the future like that. (or just add some comments about them and what happens here)

Reply:
the sync conf will be moved to configuration, ill open a task for the other ones